### PR TITLE
Introduce nix

### PR DESCRIPTION
**Pull request by blackheaven**
(to be merged after #16

### CODE REVIEW

**Comment by frasertweedale** [comment]
@blackheaven #16 was merged.

File: code/hsec-tools/flake.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+          sources = { };
+
+          jailbreakUnbreak = pkg:
+            pkgs.haskell.lib.doJailbreak (pkgs.haskell.lib.dontCheck (pkgs.haskell.lib.unmarkBroken pkg));
+
+          haskellPackages = pkgs.haskell.packages.ghc925.override
+            {
+              overrides = hself: hsuper: { };
+            };
```
[the remark this answers is not in the document]
Comment (blackheaven) [reply code/hsec-tools/flake.nix:26]: It's used twice below

**Comment by blackheaven** [comment]
> Should we document how to update the `cabal.project.freeze`?

shall I add a README for the tool?

File: code/hsec-tools/hsec-tools.cabal [hunk 2]
```diff
@@ -1,6 +1,7 @@
 cabal-version:      2.4
 name:               hsec-tools
 version:            0.1.0.0
+tested-with:        GHC==9.2.7, GHC==9.0.2, GHC==8.10.7
```
Comment (frasertweedale) [inline code/hsec-tools/hsec-tools.cabal:4]: We already have the tested-with clause below :)

**Comment by blackheaven** [comment]
let me know if there's other changes to be made, otherwise, I'll squash my fixups.

File: code/hsec-tools/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+`hesc-tools` aims to support [Haskell advisories database](https://github.com/haskell/security-advisories).
+
+## Building
+
+We aim to support both regular cabal-based and nix-based builds.
+
+## Developping
+
+Whenever you add or change denpendencies, update cabal freeze file:
```
Comment (frasertweedale) [inline code/hsec-tools/README.md:11]: typo.  `denpendencies` -> `dependencies`.

File: code/hsec-tools/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# hsec-tools
+
+`hesc-tools` aims to support [Haskell advisories database](https://github.com/haskell/security-advisories).
+
+## Building
+
+We aim to support both regular cabal-based and nix-based builds.
+
+## Developping
```
Comment (frasertweedale) [inline code/hsec-tools/README.md:9]: typo. `Developping` -> `Developing`

**Comment by frasertweedale** [comment]
> let me know if there's other changes to be made, otherwise, I'll squash my fixups.

There are a couple of typographic errors.  Sorry I didn't notice these during my first review!

**Comment by chessai** [comment]
> > let me know if there's other changes to be made, otherwise, I'll squash my fixups.
> 
> There are a couple of typographic errors.  Sorry I didn't notice these during my first review!

FYI, GitHub has a very useful `suggestion` feature, so you as a reviewer can comment a suggested change (very useful for typos), and the PR owner can commit the changes directly in the UI, either as singletons or in a batch.

**Comment by blackheaven** [comment]
> 

No worries, I've just added it.

I use fixups to keep review history, it does not aim to be merge as-it.

**Comment by frasertweedale** [comment]
LGTM.  Feel free to squash and merge.

**Pull request merged**